samples/secure_tunneling/secure_tunnel/main.cpp

```diff
@@ -0,0 +1,278 @@
+    }
+
+    /*
+     * localProxyMode is set to destination by default unless flag is set to source
+     */
+    if (s_cmdOptionExists(argv, argv + argc, "--localProxyModeSource"))
+    {
+        localProxyMode = AWS_SECURE_TUNNELING_SOURCE_MODE;
+    }
+    else
+    {
+        localProxyMode = AWS_SECURE_TUNNELING_DESTINATION_MODE;
+    }
+
+    if (s_cmdOptionExists(argv, argv + argc, "--message"))
+    {
+        message = s_getCmdOption(argv, argv + argc, "--message");
+    }
+
+    /*
+     * You need an event loop group to process IO events.
+     * If you only have a few connections, 1 thread is ideal
+     */
+    Io::EventLoopGroup eventLoopGroup(1);
+    if (!eventLoopGroup)
+    {
+        fprintf(
+            stderr, "Event Loop Group Creation failed with error %s\n", ErrorDebugString(eventLoopGroup.LastError()));
+        exit(-1);
+    }
+
+    Io::DefaultHostResolver defaultHostResolver(eventLoopGroup, 2, 30);
+    Io::ClientBootstrap bootstrap(eventLoopGroup, defaultHostResolver);
+
+    if (!bootstrap)
+    {
+        fprintf(stderr, "ClientBootstrap failed with error %s\n", ErrorDebugString(bootstrap.LastError()));
+        exit(-1);
+    }
+
+    /*********************** Callbacks ***************************/
+    auto OnConnectionComplete = [&]() {
+        switch (localProxyMode)
+        {
+            case AWS_SECURE_TUNNELING_DESTINATION_MODE:
+                fprintf(stdout, "Connection Complete in Destination Mode\n");
+                break;
+            case AWS_SECURE_TUNNELING_SOURCE_MODE:
+                fprintf(stdout, "Connection Complete in Source Mode\n");
+                fprintf(stdout, "Sending Stream Start request\n");
+                mSecureTunnel->SendStreamStart();
+                break;
+        }
+    };
+
+    auto OnConnectionShutdown = [&]() {
+        fprintf(stdout, "Connection Shutdown\n");
+        fprintf(stdout, "Sample Complete");
+        exit(0);
+    };
+
+    auto OnSendDataComplete = [&](int error_code) {
+        switch (localProxyMode)
+        {
+            case AWS_SECURE_TUNNELING_DESTINATION_MODE:
+                if (!error_code)
+                {
+                    fprintf(stdout, "Send Data Complete in Destination Mode\n");
+                }
+                else
+                {
+                    fprintf(stderr, "Send Data Failed with error code %d\n", error_code);
+                }
+
+                break;
+            case AWS_SECURE_TUNNELING_SOURCE_MODE:
+                if (!error_code)
+                {
+                    fprintf(stdout, "Send Data Complete in Source Mode\n");
+                }
+                else
+                {
+                    fprintf(stderr, "Send Data Failed with error code %d\n", error_code);
+                }
+                break;
+        }
+    };
+
+    auto OnDataReceive = [&](const struct aws_byte_buf &data) {
+        receivedData = std::string((char *)data.buffer, data.len);
```

Comment:
A little bit weird to have declared `receivedData` outside this function. Just make it a local variable

I think I already said that...